functions/pupil-status/pupil-status.service.js

```diff
@@ -69,6 +69,33 @@ async function changePupilState (pupilId, targetStatusCode) {
   return sqlService.modify(sql, params)
 }
 
+/**
+ * Filter for live checks and make a request for the pupil-status to be updated for multiple pupils on v2
+ * @param {Object} logger
+ * @param {String} logPrefix
+ * @param {[{checkId: <number>, pupilId: <number>, checkCode: <string>, isLiveCheck: <boolean>}]} checkData - must contain `checkId`, `pupilId`, `checkCode` and `isLiveCheck` props
+ * @return {Promise<*|Promise<*>>}
+ */
+async function updatePupilStatusForLiveChecksV2 (logger, logPrefix, checkData) {
+  logger.info(`${logPrefix}: updatePupilStatusV2(): got ${checkData.length} pupils`)
```

Comment:
How do we know checkData is defined and an Array?

Reply:
updated